src/importexport/tabledit/internal/measurehandler.cpp

```diff
@@ -130,25 +132,28 @@ int MeasureHandler::measureIndex(int tstart, const std::vector<TefMeasure>& tefM
             return i;
         }
     }
-    return -1; // not found
+    return muse::nidx; // not found
 }
 
 // return the offset of tstart (distance from its measure's start)
 
 int MeasureHandler::offsetInMeasure(int tstart, const std::vector<TefMeasure>& tefMeasures)
 {
-    auto index { measureIndex(tstart, tefMeasures) };
-    if (0 <= index) {
-        return tstart - nominalMeasureStarts.at(index);
+    size_t index{ measureIndex(tstart, tefMeasures) };
+    if (index >= tefMeasures.size()) {
+        return -1; // not found
     }
-    return -1; // not found
+    return tstart - nominalMeasureStarts.at(index);
 }
 
 // find the smallest offset of any note in a pickup measure
 
 void MeasureHandler::updateGapLeft(std::vector<int>& gapLeft, const int position, const std::vector<TefMeasure>& tefMeasures)
 {
-    auto index { measureIndex(position, tefMeasures) };
+    size_t index{ measureIndex(position, tefMeasures) };
+    if (index >= tefMeasures.size()) {
+        return; // not found
+    }
     if (tefMeasures.at(index).isPickup) {
         auto offset { offsetInMeasure(position, tefMeasures) };
         if (0 <= index && 0 <= offset) {
```

Comment:
Oh wait, now we've got a new one here... Looks like we just need to remove `0 <= index &&`
(same below)